### PR TITLE
Enhance API responses

### PR DESCRIPTION
**Pull request by Amna-Alshawa**
## Feature description
Create Shift Permission API for HO employees to apply for Early Leave or Late Entry.
Display Roles per Employee

## Solution description
Enhance API responses.
Add status code to the response, remove method, and typo fix.

## For Shift Permission:
Created Successfully: (After adding status code)
created

Not Successfully:  (After Adding status code)
Not Created

Duplicate Record:(After Adding status code)
Duplicate

## For Return Roles API:
roles

## Areas affected and ensured
Nothing.

## Is there any existing behavior change of other features due to this code change?
No

## Was this feature tested on all the browsers?
  - [X] Chrome
  - [] Safari

### CODE REVIEW

**Comment by Amna-Alshawa** [comment]
**@Ehtesham599**, Kindly, review my PR. I have added all your suggestions, please check once you have time.

File: one_fm/api/mobile/shift_permission.py [hunk 1]
```diff
@@ -11,46 +11,76 @@ def create_shift_permission(employee, permission_type, date, reason, leaving_tim
...
+                doc = frappe.get_doc('Shift Permission',{"employee": employee, "date":date, "assigned_shift": assigned_shift, "permission_type": permission_type})
+                return response({'message':"{0} has already applied for permission to {1} on {2}.".format(doc.emp_name, permission_type.lower(), date),'data':{}},409)
 
+        elif not shift or not type or not assigned_shift or not shift_supervisor:
+            return response({'message':"Shift Details are missing, Please Make sure You are have a Scheduled record on {0}".format(date),'data':{}},400)
+           
     except Exception as e:
         frappe.log_error(frappe.get_traceback())
-        return frappe.utils.response.report_error(e)
+        return response({'message':"Shift Permission Not Created Successfully",'data':{}},400)
```
Comment (Ehtesham599) [inline one_fm/api/mobile/shift_permission.py]: The status code should be set as 500 and not 400 for an internal server error.

File: one_fm/api/mobile/shift_permission.py [hunk 2]
```diff
@@ -11,46 +11,76 @@ def create_shift_permission(employee, permission_type, date, reason, leaving_tim
...
+    return response({'messages':"Roles Are Successfully Listed ",'data':user_roles},200)
+
+# This method returing the message and status code of the API
+def response(message, status_code):
+    """
+    Params: message, status code
+    """
+    frappe.local.response["message"] = message
+    frappe.local.response["http_status_code"] = status_code
     return user_roles
```
Comment (Ehtesham599) [inline one_fm/api/mobile/shift_permission.py]: Why are you returning user_roles here?

**Review by Ehtesham599 — CHANGES_REQUESTED** [review]

File: one_fm/api/mobile/shift_permission.py (hunk 2 above)
Comment (Amna-Alshawa) [reply one_fm/api/mobile/shift_permission.py]: By mistake while resolving the conflict.

File: one_fm/api/mobile/shift_permission.py (hunk 1 above)
Comment (Amna-Alshawa) [reply one_fm/api/mobile/shift_permission.py]: Noted

File: one_fm/api/mobile/shift_permission.py [hunk 3]
```diff
@@ -11,47 +11,77 @@ def create_shift_permission(employee, permission_type, date, reason, leaving_tim
     Return: shift permission record
 	"""
     try:
-        shift,type,assigned_shift,shift_supervisor = get_shift_details(employee,date)
+        shift, type, assigned_shift, shift_supervisor = get_shift_details(employee,date)
         if shift and type and assigned_shift and shift_supervisor:
-            doc = frappe.new_doc('Shift Permission')
-            doc.employee = employee
```
Comment (Amna-onefm) [inline one_fm/api/mobile/shift_permission.py:17]: testing comments

**Comment by Amna-Alshawa** [comment]
@Ehtesham599, Resolved conversations and thank you for your updates.

**Review by Ehtesham599 — APPROVED** [review]

**Pull request merged**